run_splunk_container.sh

```diff
@@ -0,0 +1,39 @@
+#!/bin/bash
+
+source ./preview.env
+
+SPLUNK_PASSWORD="Password"
+SPLUNK_START_ARGS="'--accept-license'"
+URL="http://localhost:8089"
+
+build_and_run_image() {
+    # Ensure the IMAGE_NAME exists
+    [[ -z "$IMAGE_NAME" ]] && echo "Error: IMAGE_NAME ENV is not set!" && exit 1
+
+    echo "Building Podman image..."
+    podman build -t "$IMAGE_NAME" . && echo "Image '$IMAGE_NAME' built successfully!" || { echo "Error: Failed to build the image."; exit 1; }
+
+    echo "Running Podman container..."
+    podman run -d -e SPLUNK_PASSWORD="$SPLUNK_PASSWORD" -e SPLUNK_START_ARGS="$SPLUNK_START_ARGS" -p 8000:$WEB_PORT -p 8089:$API_PORT "$IMAGE_NAME" && echo "Container from image '$IMAGE_NAME' started successfully!" || { echo "Error: Failed to start the container."; exit 1; }
+}
+
+wait_for_splunk_cli() {
```

Comment:
Perhaps I'm missing something but it looks like this waits for the splunk API at `http://localhost:8089` to respond with a `200` status code. The name of this function suggests something different.

Reply:
fter the ansible playbook within splunk is completed. we are restarting the splunk. this is to check if splunk is accessible or not. I will change the name appropriately